### PR TITLE
Add Commit Hash to prow-tests images

### PR DESCRIPTION
**Pull request by upodroid**
**What this PR does, why we need it**:

**Which issue(s) this PR fixes**:
Part of #3023

**Special notes to reviewers**:

**User-visible changes in this PR**:

/kind enhancement
/assign @chizhg

### CODE REVIEW

File: images/prow-tests/Dockerfile [hunk 1]
```diff
@@ -15,9 +15,12 @@
 FROM gcr.io/k8s-testimages/kubekins-e2e:v20210113-cc576af-master AS stable-stuff
 
 ENV DEBIAN_FRONTEND noninteractive
+ARG COMMIT_HASH
 ARG GCLOUD_VERSION=368.0.0
 ARG KUBECTL_VERSION=v1.21.4
 
+RUN echo "${COMMIT_HASH}" > /commit_hash
```
Comment (chizhg) [inline images/prow-tests/Dockerfile:22]: I guess it's already the plan but just want to double check - after this PR is merged, are you going to add one line in https://github.com/knative/hack/blob/main/presubmit-tests.sh#L252-L283 to print out the commit hash?

**Review by chizhg — COMMENTED** [review]
/lgtm
/approve

File: images/prow-tests/Dockerfile (hunk 1 above)
Comment (upodroid) [reply images/prow-tests/Dockerfile:22]: that is right

**Pull request merged**